src/gui/attributetable/qgsdualview.cpp

```diff
@@ -85,6 +85,8 @@ void QgsDualView::init( QgsVectorLayer* layer, QgsMapCanvas* mapCanvas, const Qg
   else
     columnBoxInit();
 
+  mTableView->resizeColumnsToContents();
```

Comment:
@m-kuhn yes, the best solution would probably be adjusting on visible rows, unfortunately I don't know when I'll have some more time to investigate on this. For the time being it's better to delete that line (maybe leaving a TODO)